httpx/_client.py

```diff
@@ -933,7 +933,7 @@ def _send_handling_auth(
                     request = next_request
                     history.append(response)
 
-                except Exception as exc:
+                except BaseException as exc:
```

Comment:
Maybe catch `Exception` and `asyncio.exceptions.CancelledError`?

Reply:
Here we need catch all exceptions to ensure response closed, just like `async with` context manager. And the exception will re-raised so it will not swallow errors.